cookieConsent/OneTrust/browser.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable class-methods-use-this */
+class OneTrust {
+  constructor(sourceConfig) {
+    this.sourceConfig = sourceConfig;
+  }
+
+  isEnabled(destConfig) {
+    // If user does not load onetrust sdk before loading rudderstack sdk
+    // we will not be filtering any of the destinations.
+
+    if (!window.OneTrust && !window.OnetrustActiveGroups) {
+      return true;
+    }
+
+    /**
+     * Structure of onetrust consent group destination config.
+     * 
+     * "oneTrustConsentGroup": [
+                        {
+                            "oneTrustConsentGroup": "Performance Cookies"
+                        },
+                        {
+                            "oneTrustConsentGroup": "Functional Cookies"
+                        },
+                        {
+                            "oneTrustConsentGroup": ""
+                        }
+                    ]
+      *
+     */
+
+    const { oneTrustConsentGroup } = destConfig; // mapping of the destination with the consent group name
+
+    // If the destination do not have this mapping events will be sent.
+
+    if (!oneTrustConsentGroup) {
```

Comment:
AFAIK, we'll get the following when no category is set,

```
"oneTrustConsentGroup": [
  {
    "oneTrustConsentGroup":  ""
  }
]
```

Verify and update accordingly.

Reply:
yeah and thus filtering later and checking the array